cmd/project.go

```diff
@@ -38,6 +38,38 @@ Use the subcommands to manage your projects.`,
 	},
 }
 
+// detectPackageManager auto-detects the package manager based on lockfiles in the directory
+// If no lockfile is found, falls back to the default runtime from the template
+func detectPackageManager(dir string, defaultRuntime string) string {
+	// Only auto-detect for JavaScript-based runtimes
+	if defaultRuntime != "nodejs" && defaultRuntime != "bunjs" && defaultRuntime != "pnpm" {
+		return defaultRuntime
+	}
+
+	// Check for pnpm lockfile first (most specific)
+	if util.Exists(filepath.Join(dir, "pnpm-lock.yaml")) {
+		return "pnpm"
```

Comment:
we can't return this as this isn't a valid runtime. this object will get serialized and used on the backend to run the correct runtime.

we could break out the function to use for running the module install from setting the runtime